application-analytics-default/src/main/java/com/xwiki/analytics/internal/RowEvolutionJsonNormaliser.java

```diff
@@ -0,0 +1,120 @@
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+
+/**
+ * Normalizes the response needed by the RowEvolution feature.
+ *
+ * @version $Id$
+ * @since 1.0
+ */
+@Component
+@Named("RowEvolution")
+@Unstable
+@Singleton
+public class RowEvolutionJsonNormaliser extends AbstractJsonNormaliser
+{
+    /**
+     * Hint for the RowEvolution.
+     */
+    public static final String HINT = "RowEvolution";
+
+    private static final String DATE = "date";
+
+    @Override
+    public String getIdentifier()
+    {
+        return RowEvolutionJsonNormaliser.HINT;
+    }
+
+    /**
+     * Transforming the json object returned by Matomo into an array of jsons to make it easier to use in javascript and
+     * adding a date to each entry to be able to idenitfy for what period the statistics are.
+     *
+     * @param jsonNode the response from Matomo that is a json
+     * @param filters holds the criteria for filtering a dataset
+     * @return filtered array of processed jsons with the date added to them
+     */
+    protected ArrayNode processObjectNode(JsonNode jsonNode, Map<String, String> filters) throws JsonProcessingException
+    {
+        ArrayNode arrayNode = OBJECT_MAPPER.createArrayNode();
+        Iterator<String> fieldNames = jsonNode.fieldNames();
+        Map<String, String> processingValues = new HashMap<>();
+        while (fieldNames.hasNext()) {
+            String date = fieldNames.next();
+            processingValues.put(DATE, date);
+            JsonNode childNode = jsonNode.get(date);
+            boolean nodeFound = true;
+            for (JsonNode node : childNode) {
+                if (matchesAllFilters(node, filters)) {
+                    arrayNode.add(this.processNode(node, processingValues));
+                    nodeFound = false;
+                    break;
+                }
+            }
+            // When the node is empty or the entry wasn't found then we create a new node with the current date and
+            // add it to the array.
+            if (childNode.get(0) == null || nodeFound) {
+                arrayNode.add(processNode(OBJECT_MAPPER.createObjectNode(), processingValues));
+
+            }
+        }
+        return arrayNode;
+    }
+
+    @Override
+    protected JsonNode processNode(JsonNode currentNode, Map<String, String> processingValues)
+    {
+        ((ObjectNode) currentNode).put(DATE, processingValues.get(DATE));
+        return currentNode;
+    }
+
+    @Override
+    protected boolean matchesAllFilters(JsonNode objNode, Map<String, String> filters)
+    {
+        if (filters == null) {
+            return true;
+        }
+        for (Map.Entry<String, String> entry : filters.entrySet()) {
+            String filterField = entry.getKey();
+            String filterValue = entry.getValue();
+            // Exact match instead of partial match like in the original method.
```

Comment:
`An exact match is needed instead of the partial one, because row evolution is done on a specific entry.`